optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,474 @@
+                `transformers.file_utils.default_cache_path`, which is the cache directory for transformers.
+            config (`PretrainedConfig`) :
+                an object of PretrainedConfig.
+            use_auth_token (`str` or `bool`):
+                Is needed to load models from a private repository
+            revision (`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization
+        """
+        if is_torch_version("<", "2.0.0"):
+            raise ImportError("`torch>=2.0.0` is needed to trace your model")
+
+        task = cls.export_feature
+        model_kwargs = {
+            "revision": revision,
+            "use_auth_token": use_auth_token,
+            "cache_dir": cache_dir,
+            "subfolder": subfolder,
+            "local_files_only": local_files_only,
+            "force_download": force_download,
+            "torch_dtype": torch_dtype,
+        }
+
+        if config.torch_dtype != "int8" and config.torch_dtype != torch.int8:
+            model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
+        else:
+            file_name = kwargs.get("file_name", WEIGHTS_NAME)
+            # Load the model from local directory
+            if os.path.isdir(model_id):
+                state_dict_path = os.path.join(model_id, file_name)
+            # Download the model from the hub
+            else:
+                state_dict_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+            # Load the state dictionary of the model to verify whether the model is quantized or not
+            state_dict = torch.load(state_dict_path, map_location="cpu")
+            if "best_configure" in state_dict and state_dict["best_configure"] is not None:
+                subfolder = kwargs.get("subfolder", "")
+                arg_framework = kwargs.get("framework", None)
+                framework = TasksManager.determine_framework(model_id, subfolder=subfolder, framework=arg_framework)
+                model_class = TasksManager.get_model_class_for_task(task, framework)
+                init_contexts = [no_init_weights(_enable=True)]
+                with ContextManagers(init_contexts):
+                    model = model_class.from_pretrained(model_id, **model_kwargs)
+                try:
+                    model = load(state_dict_path, model)
+                except Exception as e:
+                    logger.error(e.args)
+                    raise
+            else:
```

Comment:
I think we should raise an error in this case as it means the quantized model will not be loaded correctly

Reply:
Done